File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.py
class Solution(object):
    def intersection(self, nums1, nums2):
        nums1 = set(nums1)
        result = []
        for num in nums2:
            if num in nums1:
                result.append(num)
                nums1.remove(num)
        return result

        # return list(set(nums1).intersection(set(nums2)))

File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.java
class Solution 
{
    public int[] intersection(int[] nums1, int[] nums2) 
    {
        Set<Integer> set = new HashSet<>();
        List<Integer> res = new ArrayList<>();
        for(int num:nums2)
            set.add(num);
        for(int num:nums1)
        {
            if(set.contains(num))
            {
                res.add(num);
                set.remove(num);
            }
        }
        int[] ans=new int[res.size()];
        int i=0;
        for(int num:res)
            ans[i++]=num;
        return ans;
    }
}